scripts/asr_language_modeling/ngram_lm/kenlm_utils.py

```diff
@@ -17,72 +17,168 @@
 Utility methods to be used for training N-gram LM with KenLM in train_kenlm.py
 """
 
+import argparse
+import gzip
 import json
 import os
+import re
 
 import numpy as np
+import torch
 from joblib import Parallel, delayed
 from tqdm.auto import tqdm
 
+import nemo.collections.asr as nemo_asr
+from nemo.collections.asr.parts.submodules.ctc_beam_decoding import DEFAULT_TOKEN_OFFSET
+from nemo.collections.asr.parts.utils.transcribe_utils import PunctuationCapitalization, TextProcessor
+from nemo.collections.common.tokenizers.sentencepiece_tokenizer import SentencePieceTokenizer
+from nemo.utils import logging
+
 # List of the supported models to be used with N-gram LM and beam search decoding
 SUPPORTED_MODELS = {
     'EncDecCTCModelBPE': 'subword',
     'EncDecCTCModel': 'char',
     'EncDecRNNTBPEModel': 'subword',
     'EncDecRNNTModel': 'char',
 }
+CHUNK_SIZE = 8192
+CHUNK_BUFFER_SIZE = 512
 
 
 def softmax(x):
     e = np.exp(x - np.max(x))
     return e / e.sum(axis=-1).reshape([x.shape[0], 1])
 
 
-def read_train_file(path, lowercase: bool = False):
+def get_train_list(args_train_path):
+
+    train_path = []
+    for train_item in args_train_path:
+        if os.path.isdir(train_item):
+            file_list = os.listdir(train_item)
+            train_path.extend([os.path.join(train_item, file) for file in file_list])
+
+        elif os.path.isfile(train_item):
+            train_path.append(train_item)
+    return sorted(train_path)
+
+
+def setup_tokenizer(tokenizer_model_file):
+    """ TOKENIZER SETUP """
+    logging.info(f"Loading nemo model '{tokenizer_model_file}' ...")
+    if tokenizer_model_file.endswith('.model'):
+        tokenizer_nemo = SentencePieceTokenizer(tokenizer_model_file)
```

Comment:
Don't assume tokenizer. Get from model.

Reply:
need it to use one tokenizer from many in NeMo model (in case of AggregateTokenizer)